gnome-initial-setup/pages/language/cc-language-chooser.c

```diff
@@ -145,10 +148,63 @@ language_widget_new (const char *locale_id,
         return widget->box;
 }
 
+static inline gint
+get_selected_language_row_y (GtkWidget *row)
+{
+        GtkAllocation alloc;
+
+        gtk_widget_get_allocation (row, &alloc);
+
+        return alloc.y + alloc.height / 2.0;
+}
+
+static void
+update_scroll_position (CcLanguageChooser *self)
+{
+        GtkAdjustment *vadjustment;
+        GtkWidget *row;
+        gdouble page_increment;
+        gdouble real_value;
+
+        row = self->priv->to_be_scrolled_row;
+
+        vadjustment = gtk_scrolled_window_get_vadjustment (GTK_SCROLLED_WINDOW (self->priv->language_scrolledwindow));
+        page_increment = gtk_adjustment_get_page_increment (vadjustment);
+        real_value = get_selected_language_row_y (row) - page_increment / 2.0;
+
+        /*
+         * Since GtkScrolledWindow tries to follow the focused row of a listbox,
+         * make sure that the current row is focused.
+         */
+        gtk_widget_grab_focus (row);
+
+        gtk_adjustment_set_value (vadjustment, real_value);
+
+        self->priv->initial_scroll = TRUE;
+}
+
+static void
+schedule_scroll (CcLanguageChooser *self,
+                 GtkWidget         *row)
+{
+        self->priv->to_be_scrolled_row = row;
+
+        if (gtk_widget_get_realized (self->priv->language_scrolledwindow)) {
+                update_scroll_position (self);
+                return;
+        } else {
```

Comment:
Sorry for being so pedantic here, but if we return early, we should also get rid of the `else`